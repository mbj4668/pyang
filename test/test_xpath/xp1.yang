module xp1 {
  yang-version 1.1;
  namespace urn:xp1;
  prefix xp1;

  container a {
    list b {
      key id;
      leaf id {
        type string;
      }
      leaf type {
        type string;
      }
      container c {
        container d {
          leaf e1 {
            when 'ancestor::xp1:b/xp1:type = "bar"';
            type int32;
          }
          leaf e2 {
            when 'ancestor::b/type = "bar"';
            type int32;
          }
          leaf e3 {
            when 'ancestor-or-self::b/type = "bar"';
            type int32;
          }
          leaf f {
            when 'ancestor::xp1:b/typex = "bar"'; // error
            type int32;
          }
          leaf g1 {
            when 'ancestor::xp1:bx/type = "bar"'; // error
            type int32;
          }
          leaf g2 {
            when 'ancestor-or-self::xp1:bx/type = "bar"'; // error
            type int32;
          }
          leaf h {
            when 'ancestor::node()/xp1:b/typex = "bar"'; // can't check node()
            type int32;
          }
        }
        list b {
          key name;
          leaf name {
            type string;
          }
          leaf x {
            when 'ancestor::xp1:b/xtype = "bar"'; // NOTE: this is an error, but we have
                                                  // multiple ancestors called "b", and the code can't
                                                  // handle that
            type int32;
          }
        }
      }
    }
  }
}
