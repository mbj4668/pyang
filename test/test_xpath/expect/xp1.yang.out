xp1.yang:30: warning: XPATH_NODE_NOT_FOUND1
xp1.yang:34: warning: XPATH_ANCESTOR_NOT_FOUND
xp1.yang:38: warning: XPATH_ANCESTOR_NOT_FOUND
xp1.yang:52: warning: XPATH_MULTIPLE_ANCESTORS
